src/System.Runtime/tests/System/HashCodeTests.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Xunit;
+
+public static class HashCodeTests
+{
+
+#   if SYSTEM_HASHCODE_TESTVECTORS
+    [Theory]
+    [InlineData(0x02cc5d05U)]
+    [InlineData(0xa3643705U, 0x64636261U)]
+    [InlineData(0x4603e94cU, 0x64636261U, 0x33323130U)]
+    [InlineData(0xd8a1e80fU, 0x64636261U, 0x33323130U, 0x68676665U)]
+    [InlineData(0x4b62a7cfU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U)]
+    [InlineData(0xc33a7641U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U)]
+    [InlineData(0x1a794705U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U)]
+    [InlineData(0x4d79177dU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU)]
+    [InlineData(0x59d79205U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U)]
+    [InlineData(0x49585aaeU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U)]
+    [InlineData(0x2f005ff1U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U)]
+    [InlineData(0x0ce339bdU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U)]
+    [InlineData(0xb31bd2ffU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U, 0x33323130U)]
+    [InlineData(0xa821efa3U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U, 0x33323130U, 0x62617a79U)]
+    public static void HashCode_Add(uint expected, params uint[] vector)
+    {
+        var hc = new HashCode();
+        for (var i = 0; i < vector.Length; i++)
+            hc.Add(vector[i]);
+
+        Assert.Equal(expected, unchecked((uint)hc.ToHashCode()));
+    }
+
+    [Fact]
+    public static void HashCode_Add_Generic()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(100.2);
```

Comment:
I'd rather avoid using floats. If this test starts failing on some platform due to floating point precision differences, it'll be awful to debug.

Reply:
This should be OK in this case, as the CLR uses IEEE floats, the float is loaded from a literal, the spec says that conversion from R8 to internal float and back results in the same value, and GetHashCode() for floats should only use the bits available from the 32/64 bit of IEEE float, not any platform specific extended precision bits.

But, on the other hand, for testing, it would likely be more interesting to use a non-primitive value than a floating point number type.